src/WebJobs.Extensions.DurableTask/Options/AzureStorageOptions.cs

```diff
@@ -144,15 +145,31 @@ public void ValidateHubName(string hubName)
 
         private static string GetTaskHubErrorString(string hubName)
         {
-            return $"Task hub name '{hubName}' should contain only alphanumeric characters and have length between {MinTaskHubNameSize} and {MaxTaskHubNameSize}.";
+            return $"Task hub name '{hubName}' should contain only alphanumeric characters, start with a letter, and have length between {MinTaskHubNameSize} and {MaxTaskHubNameSize}.";
         }
 
         internal bool IsSanitizedHubName(string hubName, out string sanitizedHubName)
         {
-            sanitizedHubName = new string(hubName.ToCharArray()
-                                .Where(char.IsLetterOrDigit)
+            // Only alphanumeric characters are valid.
+            var validHubNameCharacters = hubName.ToCharArray().Where(char.IsLetterOrDigit);
+
+            if (!validHubNameCharacters.Any())
+            {
+                sanitizedHubName = TaskHubPadding;
```

Comment:
What is this condition about? If there are no valid characters, we name the task hub "Hub"? I find your use of the `TaskHubPadding` constant here really confusing since you're not using it to pad anything.

Reply:
That's fair.

I think in the case where there are no valid characters (i.e. someone names their site `------------`), we should just default to something simple like "DefaultTaskHub".